File: testsuite/matlab_and_octave/util/fix_kernel_name_inconsistency.m
function y = fix_kernel_name_inconsistency (kname)
	kname=toupper(kname);
	if findstr('SIMPLELOCALITYIMPROVEDSTRING', kname)
		y='SLIK';
	elseif findstr('LOCALITYIMPROVEDSTRING', kname)
		y='LIK';
	elseif findstr('WORDMATCH', kname)
		y='MATCH';
	elseif findstr('WEIGHTEDDEGREEPOSITIONSTRING', kname)
		y='WEIGHTEDDEGREEPOS';
	elseif strcmp(kname, 'WEIGHTEDCOMMWORDSTRING')==1
		y='WEIGHTEDCOMMSTRING';
	elseif findstr('COMMULONGSTRING', kname)
		y='COMMSTRING';
	elseif findstr('COMMWORDSTRING', kname)
		y='COMMSTRING';
	elseif findstr('STRING', kname)
		pos=findstr('STRING', kname);
		y=kname(1:pos-1);
	elseif findstr('WORD', kname)
		pos=findstr('WORD', kname);
		y=kname(1:pos-1);
	else
		y=kname;
	end

File: tests/regression/r_static/util/fix_kernel_name_inconsistency.R
fix_kernel_name_inconsistency = function(kname) {
	kname=toupper(kname);

	if (regexpr('SIMPLELOCALITYIMPROVEDSTRING', kname)>0) {
		return('SLIK')
	}
	else if (regexpr('LOCALITYIMPROVEDSTRING', kname)>0) {
		return('LIK')
	}
	else if (regexpr('WORDMATCH', kname)>0) {
		return('MATCH')
	}
	else if (regexpr('WEIGHTEDDEGREEPOSITIONSTRING', kname)>0) {
		return('WEIGHTEDDEGREEPOS')
	}
	else if (regexpr('WEIGHTEDCOMMWORDSTRING', kname)>0) {
		return('WEIGHTEDCOMMSTRING')
	}
	else if (regexpr('COMMULONGSTRING', kname)>0) {
		return('COMMSTRING')
	}
	else if (regexpr('COMMWORDSTRING', kname)>0) {
		return('COMMSTRING')
	}
	else if (regexpr('WORDSTRING', kname)>0) {
		return(strsplit(kname, 'WORDSTRING')[[1]])
	}
	else if (regexpr('STRING', kname)>0) {
		return(strsplit(kname, 'STRING')[[1]])
	}
	else if (regexpr('WORD', kname)>0) {
		return(strsplit(kname, 'WORD')[[1]])
	}
	else {
		return(kname);
	}

}
